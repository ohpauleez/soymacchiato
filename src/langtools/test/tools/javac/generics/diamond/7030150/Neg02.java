/*
 * @test /nodynamiccopyright/
 * @bug 7030150
 * @summary Type inference for generic instance creation failed for formal type parameter
 *          check that compiler rejects bad number of explicit type-arguments
 * @compile/fail/ref=Neg02.out -XDrawDiagnostics Neg02.java
 */

class Neg02 {

    static class Foo<X> {
        <T> Foo(T t) {}
    }

    Foo<Integer> fi1 = new <String, Integer> Foo<>("");
    Foo<Integer> fi2 = new <String, Integer> Foo<Integer>("");
}
