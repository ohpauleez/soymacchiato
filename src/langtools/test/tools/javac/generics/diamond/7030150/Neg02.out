Neg02.java:15:24: compiler.err.cant.apply.diamond.1: (compiler.misc.diamond: Neg02.Foo), (compiler.misc.arg.length.mismatch)
Neg02.java:16:24: compiler.err.cant.apply.symbol.1: kindname.constructor, Foo, T, java.lang.String, kindname.class, Neg02.Foo<X>, (compiler.misc.arg.length.mismatch)
2 errors
